python/lsst/meas/astrom/anetAstrometry.py

```diff
@@ -0,0 +1,464 @@
+
+        if schema.getVersion() == 0:
+            self.distortedName = "centroid.distorted"
+            self.centroidKey = schema.addField(self.distortedName, type="PointD",
+                                           doc="centroid distorted for astrometry solver")
+            self.centroidErrKey = schema.addField(self.distortedName + ".err", type="CovPointF",
+                                           doc="centroid distorted err for astrometry solver")
+            self.centroidFlagKey = schema.addField(self.distortedName + ".flag", type="Flag",
+                                           doc="centroid distorted flag astrometry solver")
+        else:
+            self.distortedName = "astrom_distorted"
+            self.centroidXKey = schema.addField(self.distortedName + "_x", type="D",
+                                           doc="centroid distorted for astrometry solver")
+            self.centroidYKey = schema.addField(self.distortedName + "_y", type="D",
+                                           doc="centroid distorted for astrometry solver")
+            self.centroidXErrKey = schema.addField(self.distortedName + "_xSigma", type="F",
+                                           doc="centroid distorted err for astrometry solver")
+            self.centroidYErrKey = schema.addField(self.distortedName + "_ySigma", type="F",
+                                           doc="centroid distorted err for astrometry solver")
+            self.centroidFlagKey = schema.addField(self.distortedName + "_flag", type="Flag",
+                                           doc="centroid distorted flag astrometry solver")
+            self.centroidKey = Point2DKey(self.centroidXKey, self.centroidYKey)
+            self.centroidErrKey = CovarianceMatrix2fKey((self.centroidXErrKey, self.centroidYErrKey))
+ 
+        self.astrometer = None
+
+    @pipeBase.timeMethod
+    def run(self, sourceCat, exposure):
+        """!Match with reference sources and calculate an astrometric solution
+
+        \param[in] sourceCat catalog of measured sources (an lsst.afw.table.SourceCatalog)
+        \return a pipeBase.Struct with fields:
+        - matches: Astrometric matches
+        - matchMeta: Metadata for astrometric matches
+        \param[in,out] exposure Exposure to calibrate
+
+        The reference catalog actually used is up to the implementation
+        of the solver; it will be manifested in the returned matches as
+        a list of lsst.afw.table.ReferenceMatch objects (\em i.e. of lsst.afw.table.Match with
+        \c first being of type lsst.afw.table.SimpleRecord and \c second type lsst.afw.table.SourceRecord ---
+        the reference object and matched object respectively).
+
+        \note
+        The input sources have the centroid slot moved to a new column "centroid.distorted"
+        which has the positions corrected for any known optical distortion;
+        the 'solver' (which is instantiated in the 'astrometry' member)
```

Comment:
I made as few changes to the existing a.net astrometry task as I could. In my opinion it needs major cleanup, including using the new standard for reference catalog schema and handling distortion in a simpler way (using a DistortedTanWcs). But I think that work should be left for another ticket.

I tried to remove all existing use of the Astrometry class except in the a.net solver, though I may have missed a few.